WordPress/src/main/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProvider.kt

```diff
@@ -96,15 +97,17 @@ class WPCrashLoggingDataProvider @Inject constructor(
                 value == EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR
     }
 
-    override fun userProvider(): CrashLoggingUser? {
-        return accountStore.account?.let { accountModel ->
-            CrashLoggingUser(
-                userID = accountModel.userId.toString(),
-                email = accountModel.email,
-                username = accountModel.userName
-            )
-        }
-    }
+    override val user = MutableStateFlow(
+            accountStore.account?.let { accountModel ->
```

Comment:
After removing `get()`, if an instance of `WPCrashLoggingDataProvider` will persist between changing users in the app, we'll assign the incorrect user (the old one).

> Looking at what WCAndroid did, I am seeing that their solution looks more closely to the below alternative

In `WCAndroid` we keep logged in user in `crashLoggingUser` property and we observe changes via event bus (see [`onAccountChanged`](https://github.com/woocommerce/woocommerce-android/blob/trunk/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt#L49) method).

Reply:
Right.. Thanks for catching that @wzieba! 🙇‍♂️ 

I've updated the implementation in d00c8d917f356fe24ab946042668adb85af575bb & 361ef69eba9ca1b88594988cae5155fe18705ae6 to bring it inline with the implementation in WCAndroid.

I tested the app after 361ef69eba9ca1b88594988cae5155fe18705ae6 by logging in with one user and then switching to another one before creating a crash and it worked as expected. I verified that before this commit, as you said, the previous user was reported. So, I _think_ we are good 🤞

Reply:
👋 @oguzkocer and thanks @wzieba for catching [this](https://github.com/wordpress-mobile/WordPress-Android/pull/18437#discussion_r1197440540)! 🥇 

> @ParaskP7 I've applied your suggestions in https://github.com/wordpress-mobile/WordPress-Android/commit/5366bfe13f72318898457106d2642fd09d83d68f & https://github.com/wordpress-mobile/WordPress-Android/commit/467020b40e8456b7b994cfe2a626178b80d05cb5. Let me know if this is what you had in mind.

Totally! 👍 

> I've updated the implementation in https://github.com/wordpress-mobile/WordPress-Android/commit/d00c8d917f356fe24ab946042668adb85af575bb & https://github.com/wordpress-mobile/WordPress-Android/commit/361ef69eba9ca1b88594988cae5155fe18705ae6 to bring it inline with the implementation in WCAndroid.

Exactly, bringing WPAndroid inline with the WCAndroid implementation (which has been tested), I think, will serve us good here. 💯